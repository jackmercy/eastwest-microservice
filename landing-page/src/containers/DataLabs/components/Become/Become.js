import { memo, useState } from 'react';
import PropTypes from 'prop-types';
import CountUp from 'react-countup';
import VisibilitySensor from 'react-visibility-sensor';

import { Content } from '../../../components';

import useStyles from './become.styles';

const Become = ({ title, data }) => {
  const classes = useStyles();
  const [isVisible, setVisibility] = useState(false);

  const handleVisibilityChange = (status) => {
    if (status) {
      setVisibility(status);
    }
  };

  return (
    <div className={classes.root}>
      <Content className={classes.content}>
        <h2 className={classes.title} data-test="become-title">
          {title}
        </h2>
        <div className={classes.listItems} data-test="become-list-items">
          {data.map((item) => (
            <div
              key={item.key}
              className={classes.itemWrapper}
              data-test="become-item"
            >
              <h2 className={classes.itemTitle}>{item.title}</h2>
              <div className={classes.itemGroup}>
                <img
                  alt={item.title}
                  src={item.image}
                  loading="lazy"
                  className={classes.itemImage}
                />
                <p className={classes.itemNumber}>
                  <CountUp
                    end={item.number}
                    start={isVisible ? 0 : null}
                    {...item.countUpConfigs}
                  >
                    {({ countUpRef }) => (
                      <VisibilitySensor
                        onChange={handleVisibilityChange}
                        delayedCall
                      >
                        <span ref={countUpRef} />
                      </VisibilitySensor>
                    )}
                  </CountUp>
                </p>
              </div>
              <p className={classes.itemDescription}>{item.description}</p>
            </div>
          ))}
        </div>
      </Content>
    </div>
  );
};
Become.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      number: PropTypes.number.isRequired,
      countUpConfigs: PropTypes.shape({
        prefix: PropTypes.string,
        suffix: PropTypes.string,
      }).isRequired,
      description: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default memo(Become);
