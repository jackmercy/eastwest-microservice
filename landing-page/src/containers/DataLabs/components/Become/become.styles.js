import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: { backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.02) },
  content: { paddingTop: rem(80), paddingBottom: rem(24) },
  title: {
    textAlign: 'center',
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    margin: 0,
    transition: 'font-size .5s ease',
  },
  listItems: {
    marginTop: rem(64),
    boxSizing: 'border-box',
    display: 'flex',
  },
  itemWrapper: { boxSizing: 'border-box' },
  itemTitle: {
    fontWeight: 'bold',
    fontSize: rem(32),
    lineHeight: rem(44),
    margin: 0,
  },
  itemGroup: { marginTop: rem(48), display: 'flex', alignItems: 'center' },
  itemImage: {
    display: 'block',
    flex: 'none',
    width: '100%',
    maxWidth: rem(80),
    marginRight: rem(40),
  },
  itemNumber: {
    fontWeight: 'bold',
    fontSize: rem(64),
    lineHeight: rem(72),
    margin: 0,
    color: COLORS.primary,
  },
  itemDescription: {
    fontSize: rem(24),
    lineHeight: rem(32),
    marginTop: rem(48),
    marginBottom: 0,
  },

  [`@media only screen and (min-width: ${BREAKPOINTS.minLg}px)`]: {
    listItems: { marginLeft: rem(-20), marginRight: rem(-20) },
    itemWrapper: { paddingLeft: rem(20), paddingRight: rem(20) },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    listItems: { display: 'block' },
    itemWrapper: {
      textAlign: 'center',
      '& + $itemWrapper': { marginTop: rem(40) },
    },
    itemGroup: { justifyContent: 'center' },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    content: { paddingTop: rem(40), paddingBottom: rem(20) },
    title: { fontSize: rem(32), lineHeight: rem(40) },
    listItems: { marginTop: rem(32) },
    itemGroup: { marginTop: rem(24) },
    itemTitle: { fontSize: rem(24), lineHeight: rem(32) },
    itemImage: { maxWidth: rem(64), marginRight: rem(24) },
    itemNumber: { fontSize: rem(56), lineHeight: rem(64) },
    itemDescription: {
      fontSize: rem(16),
      lineHeight: rem(24),
      marginTop: rem(24),
    },
  },
});
