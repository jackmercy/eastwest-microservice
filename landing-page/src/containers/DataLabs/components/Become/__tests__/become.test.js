import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Become from '../Become';

describe('Testing DataLabs components > Become', () => {
  let wrapper;
  const props = {
    title: 'Become us',
    data: [
      {
        key: 'brand-become-01',
        title: 'title 01',
        description: 'decription 01',
        number: 45,
        image: 'image',
        countUpConfigs: { prefix: '', suffix: '+' },
      },
      {
        key: 'brand-become-02',
        title: 'title 02',
        description: 'decription 02',
        number: 145,
        image: 'image',
        countUpConfigs: { prefix: '', suffix: '+' },
      },
      {
        key: 'brand-become-03',
        title: 'title 03',
        description: 'decription 03',
        number: 245,
        image: 'image',
        countUpConfigs: { prefix: '', suffix: '+' },
      },
    ],
  };

  beforeAll(() => {
    wrapper = mount(<Become {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Become {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="become-title"]')).toHaveLength(1);
  });

  it('should have a list items', () => {
    expect(wrapper.find('[data-test="become-list-items"]')).toHaveLength(1);
    expect(wrapper.find('[data-test="become-item"]')).toHaveLength(
      props.data.length,
    );
  });
});
