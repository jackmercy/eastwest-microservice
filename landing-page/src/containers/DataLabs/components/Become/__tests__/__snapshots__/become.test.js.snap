// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DataLabs components > Become should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <Memo(Content)
    className="content-0-2-2"
  >
    <h2
      className="title-0-2-3"
      data-test="become-title"
    >
      Become us
    </h2>
    <div
      className="listItems-0-2-4"
      data-test="become-list-items"
    >
      <div
        className="itemWrapper-0-2-5"
        data-test="become-item"
        key="brand-become-01"
      >
        <h2
          className="itemTitle-0-2-6"
        >
          title 01
        </h2>
        <div
          className="itemGroup-0-2-7"
        >
          <img
            alt="title 01"
            className="itemImage-0-2-8"
            loading="lazy"
            src="image"
          />
          <p
            className="itemNumber-0-2-9"
          >
            <CountUp
              decimal="."
              decimals={0}
              delay={null}
              duration={null}
              easingFn={null}
              end={45}
              formattingFn={null}
              onEnd={[Function]}
              onPauseResume={[Function]}
              onReset={[Function]}
              onStart={[Function]}
              onUpdate={[Function]}
              prefix=""
              preserveValue={false}
              redraw={false}
              separator=""
              start={null}
              startOnMount={true}
              suffix="+"
              useEasing={true}
            >
              <Component />
            </CountUp>
          </p>
        </div>
        <p
          className="itemDescription-0-2-10"
        >
          decription 01
        </p>
      </div>
      <div
        className="itemWrapper-0-2-5"
        data-test="become-item"
        key="brand-become-02"
      >
        <h2
          className="itemTitle-0-2-6"
        >
          title 02
        </h2>
        <div
          className="itemGroup-0-2-7"
        >
          <img
            alt="title 02"
            className="itemImage-0-2-8"
            loading="lazy"
            src="image"
          />
          <p
            className="itemNumber-0-2-9"
          >
            <CountUp
              decimal="."
              decimals={0}
              delay={null}
              duration={null}
              easingFn={null}
              end={145}
              formattingFn={null}
              onEnd={[Function]}
              onPauseResume={[Function]}
              onReset={[Function]}
              onStart={[Function]}
              onUpdate={[Function]}
              prefix=""
              preserveValue={false}
              redraw={false}
              separator=""
              start={null}
              startOnMount={true}
              suffix="+"
              useEasing={true}
            >
              <Component />
            </CountUp>
          </p>
        </div>
        <p
          className="itemDescription-0-2-10"
        >
          decription 02
        </p>
      </div>
      <div
        className="itemWrapper-0-2-5"
        data-test="become-item"
        key="brand-become-03"
      >
        <h2
          className="itemTitle-0-2-6"
        >
          title 03
        </h2>
        <div
          className="itemGroup-0-2-7"
        >
          <img
            alt="title 03"
            className="itemImage-0-2-8"
            loading="lazy"
            src="image"
          />
          <p
            className="itemNumber-0-2-9"
          >
            <CountUp
              decimal="."
              decimals={0}
              delay={null}
              duration={null}
              easingFn={null}
              end={245}
              formattingFn={null}
              onEnd={[Function]}
              onPauseResume={[Function]}
              onReset={[Function]}
              onStart={[Function]}
              onUpdate={[Function]}
              prefix=""
              preserveValue={false}
              redraw={false}
              separator=""
              start={null}
              startOnMount={true}
              suffix="+"
              useEasing={true}
            >
              <Component />
            </CountUp>
          </p>
        </div>
        <p
          className="itemDescription-0-2-10"
        >
          decription 03
        </p>
      </div>
    </div>
  </Memo(Content)>
</div>
`;
